migrate get started for 6-0 (#11632)
storybookjs/storybook: docs/get-started/install.md

title: 'Install'
---

Use the Storybook CLI to install it in a single command. Run this inside your existing project’s root directory:

```sh
npx sb init
```

<div style="background-color:#F8FAFC">

> TODO: ask tom/dom how to proceed on this item

Reviewer: A few things we can emphasize, either here or later:
1. This is intended to be run on an existing app, not an empty project.
2. The script looks at your project's dependencies to infer what kind of project it is and provide the best configuration (JS vs TS, react vs create-react-app vs vue vs vue-cli etc.) and you can also force it to use a specific configuration using CLI flags

> TODO: ask tom/dom how to proceed on this item

Reviewer 2: OK, yep, so we'll not say what to do if you are just getting started then?

> TODO: ask tom/dom how to proceed on this item

Author: with the latest commit i've updated said text. Let me know if it's ok.

> If you haven’t started your project yet, we advise… ? TK (needs to be vetted)
</div>

The command above will make the following changes to your local environment:

- 📦 Install the required dependencies.
- 🛠 Setup the necessary scripts to run and build Storybook.
- 🛠 Add the default Storybook configuration.
- 📝 Add some boilerplate stories to get you started.

Check that everything worked by running:

```sh
npx storybook
```

It will start Storybook locally and output the address. Depending on your system configuration, it will automatically open the address in a new browser tab and you'll be greeted by a welcome screen. In that screen there are some noteworthy items:

> - A collection of usefull links for more in depth configuration and customization options you have at your disposal.

Author: fixed with the latest commit

> - A collection of usefull links for more in depth configuration and customization options you have at your disposal.

Reviewer: ```suggestion
- A collection of useful links for more in depth configuration and customization options you have at your disposal.
```

> - A collection of usefull links for you to expand our Storybook knowledge and get involved with ever growing Storybook community.

Author: fixed by the latest commit

> - A collection of usefull links for you to expand our Storybook knowledge and get involved with ever growing Storybook community.

Reviewer: ```suggestion
- A collection of useful links for you to expand our Storybook knowledge and get involved with ever growing Storybook community.
```

- A few example stories to get you started.

<div style="background-color:#F8FAFC">
 TODO: add required image per 6.0 doc and assets spreadsheet (Storybook Welcome screen)
</div>

#### Troubleshooting

If there's an installation problem, check the README for your framework (e.g [Storybook for React](https://github.com/storybookjs/storybook/app/react/README.md) for the detailed instructions)

<div style="background-color:#F8FAFC">

> TODO: ask dom/tom if the link is correct (basics/community/), or will be a Gatsby page.

Reviewer 3: This links here https://storybook.js.org/support/

title: 'Install'
---

Use the Storybook CLI to install it in a single command. Run this inside your existing project’s root directory:

```sh
npx sb init
```
<div style="background-color:#F8FAFC">
TODO: vet this per feedback (https://github.com/storybookjs/storybook/pull/11632#discussion_r458498081)

</div>

> Storybook needs to be installed into a project that is already configured. For instance with:

Reviewer 3: ```suggestion
Storybook needs to be installed into a project that is already setup with a framework. It will not work on an empty project. There are many ways to bootstrap an app in given framework including:
```

Reviewer 3: Make this an expandable section. The summary should read: "`sb init` is not made for empty projects"

Author: Fixed with latest commit let me know if it's ok

- 📦 [Create React App](https://reactjs.org/docs/create-a-new-react-app.html)

> - 📦 [Vue Cli](https://cli.vuejs.org/)

Reviewer 3: ```suggestion
- 📦 [Vue CLI](https://cli.vuejs.org/)
```

Author: Fixed with latest commit

> Or any other tooling available. 

Reviewer 3: Put this inside the expandable section reference above. Add it to the list instead of having it as a paragraph.

Author: Fixed with latest commit let me know if you're ok with it

> During its install process, Storybook will look into your project's dependencies and provide you with the best configuration available.

Reviewer 3: This does not match the Google doc. Readers will get lost here because we're offering them way too many choices. All of this manual config `--type` should be hidden away in the Troubleshooting section.

Author: Fixed with the latest commits. Let me know if you're ok with it.

- 📦 Install the required dependencies.
- 🛠 Setup the necessary scripts to run and build Storybook.
- 🛠 Add the default Storybook configuration.
- 📝 Add some boilerplate stories to get you started.

Check that everything worked by running:

```sh
npx storybook
```

It will start Storybook locally and output the address. Depending on your system configuration, it will automatically open the address in a new browser tab and you'll be greeted by a welcome screen. In that screen there are some noteworthy items:

- A collection of useful links for more in depth configuration and customization options you have at your disposal.
- A second set of links for you to expand your Storybook knowledge and get involved with the ever growing Storybook community.
- A few example stories to get you started.
 
![Storybook welcome screen](./example-welcome.png)

<details>
<summary>Troubleshooting</summary>

If there's an installation problem, check the README for your framework (e.g <a href="https://github.com/storybookjs/storybook/blob/next/app/react/README.md">Storybook for React</a>).

If all else fails, try asking for [help](https://storybook.js.org/support/)

</details>

> > > [Now that you installed Storybook successfully, let’s take a look at a story that was written for us](./whats-a-story.md)

Reviewer 3: Make this plain text. We're building a widget for this like in LearnStorybook.

Author: Fixed with latest commit
